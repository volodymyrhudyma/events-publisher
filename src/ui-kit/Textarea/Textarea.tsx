import React, { ChangeEvent } from 'react';
import { Wrapper } from './styled/Textarea.styled';

interface ITextareaProps {
  placeholder: string;
  rows: number;
  value: string;
  onChange: (e: ChangeEvent<HTMLTextAreaElement>) => void;
  error?: string;
}

const Textarea = (props: ITextareaProps) => <Wrapper {...props} />;

Textarea.defaultProps = {
  rows: 10,
};

export default Textarea;
