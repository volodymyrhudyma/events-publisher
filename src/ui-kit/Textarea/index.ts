import Component from './Textarea';

export default Component;
