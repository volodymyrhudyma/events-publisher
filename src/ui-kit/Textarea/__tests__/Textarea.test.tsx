import React from 'react';
import { shallow } from 'enzyme';
import Textarea from '../Textarea';

describe('Textarea component', () => {
  it('should render correctly', () => {
    const component = shallow(
      <Textarea placeholder="Placeholder" value="" onChange={jest.fn()} />
    );
    expect(component).toMatchSnapshot();
  });
});
