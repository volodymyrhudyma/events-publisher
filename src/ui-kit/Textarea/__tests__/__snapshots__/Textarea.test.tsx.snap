// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea component should render correctly 1`] = `
<styled.textarea
  onChange={[MockFunction]}
  placeholder="Placeholder"
  rows={10}
  value=""
/>
`;
